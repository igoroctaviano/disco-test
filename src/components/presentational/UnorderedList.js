import React from "react";

const UnorderedList = props => <ul>{props.items.map(props.itemRenderer)}</ul>;

export default UnorderedList;
