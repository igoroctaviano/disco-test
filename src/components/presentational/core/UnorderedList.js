import React from "react";
import PropTypes from 'prop-types';

const UnorderedList = ({ items = [], itemRenderer }) => <ul>{items.map(itemRenderer)}</ul>;

UnorderedList.propTypes = {
  items: PropTypes.array,
  itemRenderer: PropTypes.func
};

export default UnorderedList;
