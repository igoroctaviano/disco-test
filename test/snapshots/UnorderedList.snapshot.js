import React from "react";
import UnorderedList from "./../../src/components/presentational/core/UnorderedList.js";
import renderer from "react-test-renderer";
import { items, itemRenderer } from "./mocks.js";

test("renders correctly", () => {
  const tree = renderer
    .create(<UnorderedList items={items} itemRenderer={itemRenderer} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
