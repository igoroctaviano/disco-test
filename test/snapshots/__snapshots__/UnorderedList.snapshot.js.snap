// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  1
  2
  3
</ul>
`;
